0000-pre-shared-key-v2.md

```diff
@@ -234,6 +231,42 @@ The following is encrypted, as defined [above](#encryption). In the case of a re
 
 **TODO:** should the payment amount support numbers larger than UInt64s?
 
+#### Payment Chunk Response
+
+The following is encrypted, as defined [above](#encryption), and set as the data accompanying the payment chunk fulfillment.
+
+```
+(Numbers represent bytes)
+
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|T|           Payment ID          |  Seq  | Payment Rcved | Chu-|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|nk Rcved |             Application Data                        |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                       Application Data                        |
+\                                                               \
+\                                                               \
+|                       Application Data                      |E|
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Junk Data                            |
+\                                                               \
+\                                                               \
+|                          Junk Data                            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+| Field | Type | Description |
+|---|---|---|
+| Type (T) | 1-Byte UInt (UInt8) | Always `4` to indicate it is a response |
+| Payment ID | 16-Byte UInt | Unique ID from the request. Senders SHOULD ensure that the response ID matches the request |
+| Sequence | 4-Byte UInt (UInt32) | Chunk sequence number from the request. Senders SHOULD ensure that the sequence matches the request |
+| Payment Received | 8-Byte UInt (UInt64) | The amount the receiver has received thus far for the payment (the sum of fulfilled chunks) |
+| Chunk Received | 8-Byte UInt (UInt64) | The amount the receiver received for the given chunk |
+| Application Data | [OER Variable-Length Octet String](http://www.oss.com/asn1/resources/books-whitepapers-pubs/Overview%20of%20OER.pdf) | User data carried along with the payment |
+| Extensions (E) | 1-Byte UInt (UInt8) | Always `0`. Reserved for future use to indicate the presence of extension fields |
```

Comment:
Extensions must be ignored if you don't implement the logic to handle OER extensions. "Always `0`" implies that a parser should verify that the extensions byte is zero, which would actually break extensibility.